magento/magento2
# Resolved - attribute option label / value comparison on api POST requests

### Description (*)

    Unable to create attribute label if the attribute label value matches with the entity_id.

### Related Pull Requests

[#32593](https://github.com/magento/magento2/issues/32593)



### Fixed Issues (if relevant)

1. Fixes [https://github.com/magento/magento2/issues/32593](url)

### Manual testing scenarios (*)

1. Create Dropdown type Product Attribute with values required yes. Ex: M, L, XL.
2. Reindex and flush cache
3. POST request to /rest/all/V1/products/attributes/{attributeCode}/options
Payload:

`{ 
       "option": { 
               "label": "XXL", 
                "value": "", 
                "sort_order": 0, 
                 "is_default": false,
                 "store_labels": [ { "store_id": 1, "label": "XXL" } ] 
          }  
}`

4. Magento will return option ID as a response  (4 in my case)

5. If you try to call the API again with the same data, you should get the exception message, as the exception got displayed, because you're trying to create duplicate labels.

6.  Setting the new label same as the id that was returned in the step 4 ( 4 )
    Payload
    
    `{ 
       "option": { 
               "label": "4", 
                "value": "", 
                "sort_order": 0, 
                 "is_default": false,
                 "store_labels": [ { "store_id": 1, "label": "4" } ] 
          }  
    }`

7. New attribute is created and Magento will return option Id as response "5" in my case

### Questions or comments


### Contribution checklist (*)
